otel/otel-setup.go

```diff
@@ -0,0 +1,56 @@
+package otel
+
+import (
+	"context"
+	"log"
+
+	"go.temporal.io/sdk/contrib/opentelemetry"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.temporal.io/sdk/interceptor"
+)
+
+var tracingInterceptor interceptor.ClientInterceptor
```

Comment:
Would rather not have a global. I think it'd be best to just have `InitializeGlobalTracerProvider() (*sdktrace.TracerProvider, error)`, and the caller can call `.Shutdown()` on it if they want. No need for anything Temporal in there. Then the caller can just run `opentelemetry.NewTracingInterceptor(opentelemetry.TracerOptions{})` where they need, no need for that to be shared code.